run/fvde2john.py

```
@@ -0,0 +1,220 @@
+        ke_uuid, ke_tag, ke_keylen = pt[sp:sp+16], uint_to_int(pt[sp + 16:sp + 18]), uint_to_int(pt[sp + 18:sp + 20])
+        padding = pt[sp + 20:sp + 24]
+        keydata = pt[sp + 24: sp + 24 + ke_keylen]
+
+        # only tag 3 is needed for constructing the hash
+        if ke_tag == 3:
+            assert padding == b'\x00\x00\x00\x00'
+            volume_unlock_record = keydata
+
+            return volume_unlock_record
+
+    return None
+
+def get_all_partitions_of_type(partition_dict, part_type):
+    return [partition_dict[p]['start'] for p in partition_dict if partition_dict[p]['partition_type'] == guid_to_hex(part_type)]
+
+def load_plist_dict(pt):
+    # resultant pt has one extra malformed line in the xml, so we remove this.
+    plist_str = b''.join(pt.split(b'\n')[:-1]).decode()
+    d = plistlib.loads(plist_str)
+    return d
+
+# Recursive traversal - Recovery HD partition does not contain a lot of files, and so this approach is fine
+def traverse_filesystem(fs_object, target_file, path='/'):
+    for entry in fs_object.open_dir(path):
+        try:
+            if entry.info.name.name in [b'.', b'..']:
+                continue
+
+            file_path = os.path.join(path, entry.info.name.name.decode('utf-8'))
+
+            if entry.info.meta and entry.info.meta.type == pytsk3.TSK_FS_META_TYPE_REG:
+                if entry.info.name.name == target_file.encode():
+                    print(f"[+] Found file: {file_path}")
+                    file_data = recover_file(fs_object, file_path)
+
+                    # this returns to previous call i.e. the dir layer
+                    return file_data
+
+            # Traverse lower layer if entry is a dir
+            elif entry.info.meta and entry.info.meta.type == pytsk3.TSK_FS_META_TYPE_DIR:
+                file_data = traverse_filesystem(fs_object, target_file, file_path)
+
+                if file_data:
+                    return file_data
+
+        except Exception as e:
```

Catching a bare `Exception` always has the drawback of silencing other unexpected errors which could maybe be handled in a better way. Is there any more *specific* exception type suitable to be caught here instead?
Besides, should the program exit here, or is this a soft failure?